packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,198 @@
+import { Customer, PaymentSessionStatus } from "../models"
+import { MedusaContainer } from "../types/global"
+import {
+  PaymentProcessorContext,
+  PaymentProcessorError,
+  PaymentProcessorSessionResponse,
+} from "../types/payment-processor"
+
+/**
+ * The new payment service plugin interface
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
+
+  /**
+   * This method will be called once when the plugin will be registered
+   */
+  init(): Promise<void>
+
+  /**
+   * Initiate a payment session with the external provider
+   */
+  createPayment(
```

Comment:
suggestion: Rename to `initiatePayment(... ` to prevent it from getting mixed up with our own `createPayment` of our `PaymentProviderService`